### PR TITLE
Improve list item template

### LINKED ISSUE #4607: Feature Request: Improve List Item Templates
Currently, the layout of the list items in the list is left entirely in the hands of the users. I would like us to improve that by modifying the list item template, as well as supplying a few additional directives to aid how the content in an item is laid out.

What I expect the User Experience to be like:

```html
<igx-list>
    <igx-list-item *ngFor="let item of items">
        <igx-avatar igxListThumbnail>
            <igx-icon>{{ item.thumbIcon }}</igx-icon>
        </igx-avatar>
        <h4 igxListLine>{{ item.title }}</h4>
        <p igxListLine>{{ item.content }}</p>
        <div igxListAction>
            <igx-icon>{{ item.actionIcon }}</igx-icon>
        </div>
   </igx-list-item>
</igx-list>
```

The template above should yield something similar to the image bellow:

material list

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #4607

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by simeonoff** [comment]
@desig9stein Can you update the README.md file for the list.

File: projects/igniteui-angular/src/lib/list/list.component.spec.ts [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+    ListCustomLoadingComponent,
+    ListWithIgxForAndScrollingComponent,
+    TwoHeadersListComponent,
+    TwoHeadersListNoPanningComponent,
+    ListDirectivesComponent
 } from '../test-utils/list-components.spec';
 import { configureTestSuite } from '../test-utils/configure-suite';
 import { DisplayDensity, IDensityChangedEventArgs } from '../core/density';
-import { IgxForOfModule } from '../directives/for-of/for_of.directive';
+import { IgxForOfModule, IgxIconModule } from 'igniteui-angular';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/list/list.component.spec.ts]: You should not import from `igniteui-angular`.

File: projects/igniteui-angular/src/lib/test-utils/list-components.spec.ts [hunk 2]
```diff
@@ -1,172 +1,204 @@
 import { Component, ViewChild } from '@angular/core';
 import { IgxListComponent } from '../list';
-import { IgxForOfDirective } from '../directives/for-of/for_of.directive';
+import { IgxForOfDirective } from 'igniteui-angular';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/test-utils/list-components.spec.ts]: Again, don't import from `igniteui-angular`

File: projects/igniteui-angular/src/lib/list/list.component.spec.ts [hunk 3]
```diff
@@ -1,19 +1,34 @@
 import { QueryList } from '@angular/core';
 import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
-import { IgxListItemComponent } from './list-item.component';
+import { IgxListItemComponent } from 'igniteui-angular';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/list/list.component.spec.ts]: Fix import

File: projects/igniteui-angular/src/lib/list/list.component.spec.ts [hunk 4]
```diff
@@ -22,7 +38,7 @@ const LIST_CSS_CLASS = 'igx-list';
 const LIST_COMPACT_DENSITY_CSS_CLASS = 'igx-list--compact';
 const LIST_COSY_DENSITY_CSS_CLASS = 'igx-list--cosy';
 
-describe('List', () => {
+fdescribe('List', () => {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/list/list.component.spec.ts]: Remove the `fdescribe`

**Pull request merged**